sdk/core/azure-core/inc/azure/core/io/body_stream.hpp

```diff
@@ -282,4 +283,36 @@ namespace Azure { namespace Core { namespace IO {
     int64_t Length() const override;
   };
 
+  /**
+   * @brief A concrete implementation of #Azure::Core::IO::BodyStream that wraps another stream and
+   * reports progress
+   */
+  class ProgressBodyStream : public BodyStream {
+  private:
+    BodyStream& m_bodyStream;
```

Comment:
This violates our C++ guidelines. https://azure.github.io/azure-sdk/cpp_implementation.html#const-and-reference-members

Reply:
We should keep the constructor parameter as reference, but change this member type to pointer.

Reply:
done